katdal/categorical.py

```diff
@@ -236,58 +291,42 @@ def __len__(self):
         """Length operator indicates number of events produced by sensor."""
         return len(self.indices)
 
+    def _bool_per_dump(self, bool_per_value):
+        """Turn list of bools per unique value into an array of bools per dump."""
+        bool_per_event = np.atleast_1d(np.array(bool_per_value)[self.indices])
```

Comment:
I would say there are two typical categorical cases:

- a few small unique values with more events, e.g. mxxx_activity ('slew', 'track', 'slew', 'track')
- a few larger unique values with a few events, e.g. cal_product_B

This approach is informed by the former case, and doesn't really matter in the latter case where you have few of everything.

Or have I missed something?